consensus/consensus_v2.go

```diff
@@ -136,14 +136,19 @@ func (consensus *Consensus) onAnnounce(msg *msg_pb.Message) {
 		return
 	}
 
+	if blockObj.NumberU64() != recvMsg.BlockNum || recvMsg.BlockNum < consensus.blockNum {
+		utils.GetLogger().Warn("blockNum not match", "recvBlockNum", recvMsg.BlockNum, "blockObjNum", blockObj.NumberU64(), "myBlockNum", consensus.blockNum)
+		return
+	}
+
 	// check block data transactions
-	if err := consensus.VerifyHeader(consensus.ChainReader, blockObj.Header(), false); err != nil {
-		utils.GetLogInstance().Warn("onAnnounce block content is not verified successfully", "error", err)
+	if err := consensus.VerifyHeader(consensus.ChainReader, blockObj.Header(), false); err != nil && consensus.mode.Mode() != Syncing {
```

Comment:
better just first check if it's in sync mode and then ignore the VerifyHeader check.

Reply:
right, let me change the order because verify header is heavy